### ISSUE TITLE
[Rely] Add support for option matchers

### ISSUE THREAD

**Issue by jordwalke** [facebookexperimental/reason-native#148]
Sound like a good idea?

`expect.option(foo).toBeNone()`
`expect.option(foo).toBeSome(..)`

**Comment by kyldvs** [comment]
Yeah I've been missing option matchers

**Comment by bandersongit** [comment]
Thanks for filing! I think I had a note somewhere to do this myself. What API would are you guys thinking?

```
type optionMatchers('a) =  {
  /* the natural ones */
  toBeNone: unit => unit,
  toBeSome(~equals: equalsFn('a)=?, 'a) => unit,
  /* less natural, but probably useful? would be awesome if we could be smart and do intMatchers for ints, 
stringMatchers for strings, etc.), but I'm fairly sure that isn't possible? */
  some: someMatchers('a)
}

where 
type someMatchers('a) = {
  toSatisfy('a => bool),
  /*anything else polymorphic make sense here?*/
}
```

**Comment by kyldvs** [comment]
I think `toBeNone` and `toBeSome` are fine, but we should also just include `toBe`, used like: `expect.option(foo).toBe(Some("hello world"));`

**Comment by bandersongit** [comment]
Merged and will be released in 2.1.0
